### PR TITLE
Refactor features.

### PR DESCRIPTION
**Pull request by asanin-epfl**
- Move ``neuritefunc`` functions that expect neurons to ``neuronfunc``. The functions are:
``max_radial_distances, number_of_sections, number_of_neurites, number_of_bifurcations, number_of_forking_points, number_of_terminations, number_of_segments``
- Make ``neuronfunc`` to work with list of neurons besides a neuron and a neuron population.
- Name consistency among private variables.

### CODE REVIEW

File: neurom/features/neuritefunc.py [hunk 1]
```diff
@@ -179,57 +178,6 @@ def pl2(node):
     return _map_sections(pl2, neurites, neurite_type=neurite_type)
 
 
-################################################################################
-# Features returning one value per NEURON                                      #
-################################################################################
-
-def _map_neurons(fun, neurites, neurite_type):
```
Comment (asanin-epfl) [inline neurom/features/neuritefunc.py:186]: @mgeplf FYI. I saw that you added this code in 2016, so just for safety I'm notifying you.

Comment (mgeplf) [reply neurom/features/neuritefunc.py:186]: Thanks for the heads up.

Before than they were just `partial` wrapped functions; this way there could be documentation added to them that allowed for better docs/cli info, iirc. As long as the behavior is the same or better, I have no problems w/ the refactorings.

**Review by adrien-berchet — APPROVED** [review]
I think it's ok for us since we never call these features directly (as far I as know) but only with the `get()` function.
And it's cool to be able to use a list of Neurons!

**Review by arnaudon — APPROVED** [review]
moving of features from neurite to neuron would make a difference in the feature extraction part, as it classifies them according to neurite_types or neuron irrc, not sure of the implications, but it may break a few tests based on that

**Comment by asanin-epfl** [comment]
> moving of features from neurite to neuron would make a difference in the feature extraction part, as it classifies them according to neurite_types or neuron irrc

Those feature functions expect neurons as input, so it is misleading to keep them in `neuritefunc`.

**Comment by arnaudon** [comment]
yes, I agree with your change, I was just thinking of possible things that could break because of that, but it should be minor indeed, will you merge that in 2.4.0 branch as well?

**Comment by asanin-epfl** [comment]
> will you merge that in 2.4.0 branch as well?

yes, you can see the merging branch at the top, right below the PR's title.

**Comment by asanin-epfl** [comment]
2.4.0 will be released in a couple of weeks from now, no worries.

**Pull request merged**